packages/istanbul-report-modern-html/src/routing.js

```diff
@@ -0,0 +1,47 @@
+export function setLocation(
+    isReplace,
+    activeSort,
+    summarizerType,
+    activeFilters,
+    fileFilter,
+    expandedLines
+) {
+    const params = [
+        activeSort.sortKey,
+        activeSort.order,
+        summarizerType,
+        activeFilters.low,
+        activeFilters.medium,
+        activeFilters.high,
+        encodeURIComponent(fileFilter),
+        expandedLines.map(encodeURIComponent).join(',')
+    ];
+    const newUrl = `#${params.join('/')}`;
+
+    if (newUrl === location.hash) {
+        return;
+    }
+
+    window.history[isReplace ? 'replaceState' : 'pushState'](null, '', newUrl);
```

Comment:
this is interesting... can it not be done in a routing library? I mean I'm totally fine with this but it could be cleaner, no?

Reply:
I looked at a couple of routing libraries - I wanted to use it, but for this simple case, it was more effort than implementing this